internal/pkg/skuba/addons/dex.go

```diff
@@ -187,6 +192,12 @@ spec:
           - /usr/bin/caasp-dex
           - serve
           - /etc/dex/cfg/config.yaml
+        env:
+          - name: OIDC_CLIENT_SECRET
```

Comment:
This is just one of two client secrets, the gangway client secret. Could it be renamed to something like `OIDC_GANGWAY_CLIENT_SECRET`? I think the hardcoded CLI secret should also be generated and taken from an environment variable: https://github.com/SUSE/skuba/blob/f2424908c819dda0a7656c02332f8f880dffc293/internal/pkg/skuba/addons/dex.go#L152

Reply:
The hardcoded CLI secret in for `skuba auth login` and it's spread out to all unauthorized users. Therefore, all the skuba CLI should use the same client secret otherwise it would fail when verifying the client secret.

The upstream had a new solution for the static public client https://github.com/dexidp/dex/pull/1701.
But this PR focus on dex&gangway integration, if we have other concerns for public client secret, I'd suggest having another PR to address it. But in my first thought, it would have a backward-compatible issue because different users use different skuba CLI versions.

Reply:
Rename to `OIDC_GANGWAY_CLIENT_SECRET`.

Reply:
[**@jenting**](https://github.com/jenting) commented on [Jun 22, 2020, 5:24 PM PDT](https://github.com/SUSE/skuba/pull/1174#discussion_r443894423 "2020-06-23T00:24:54Z - Replied by Github Reply Comments"):
> The hardcoded CLI secret in for `skuba auth login` and it's spread out to all unauthorized users. Therefore, all the skuba CLI should use the same client secret otherwise it would fail when verifying the client secret.

Thanks for this information, I see now that this is somewhat common practice, e.g. https://community.auth0.com/t/authenticating-an-installed-cli-with-oidc-and-a-third-party-client/7594

> 
> The upstream had a new solution for the static public client https://github.com/dexidp/dex/pull/1701.
> But this PR focus on dex&gangway integration, if we have other concerns for public client secret, I'd suggest having another PR to address it. But in my first thought, it would have a backward-compatible issue because different users use different skuba CLI versions.

:+1:

Reply:
> Thanks for this information, I see now that this is somewhat common practice, e.g. https://community.auth0.com/t/authenticating-an-installed-cli-with-oidc-and-a-third-party-client/7594
> 

yes, we took google cloud SDK design as the reference when we implemented `skuba auth login`, good finding :100: .